pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -193,6 +193,11 @@ func (r *ReconcileRemoteMachineSet) syncMachineSets(
 	// we extract installer image refs.
 	var amiID string
 	for _, ms := range remoteMachineSets.Items {
+		if ms.Spec.Template.Spec.ProviderSpec.Value == nil {
+			cdLog.WithField("machineSet", ms.Name).Warn("machine set has nil ProviderSpec")
+			// keep looking for an AMI
+			continue
+		}
 		awsProviderSpec, err := decodeAWSMachineProviderSpec(ms.Spec.Template.Spec.ProviderSpec.Value, r.scheme)
 		if err != nil {
 			cdLog.WithError(err).Error("error decoding AWSMachineProviderConfig")
```

Comment:
This should probably not be an error log. It should not be the job of hive to report errors in user-configured machinesets. Also, the function should not return an error in this case. It should just skip the machineset.